fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java

```diff
@@ -287,4 +296,89 @@ public long getTableCreateTime(String dbName, String tblName) throws Exception {
         }
         return 0;
     }
+
+    public List<PartitionInfo> getChangedPartitionInfo(Table table, long latestSnapShotID) {
+        LOG.debug("Get changed partitionInfo start, table:{} ,latestSnapShotID {}", table, latestSnapShotID);
+        List<PartitionInfo> result = new ArrayList<>();
+        PaimonTable paimonTable = (PaimonTable) table;
+        Map<String, Long> partitionToSnapshotId = fetchChangedPartitionWithVersion(paimonTable, latestSnapShotID);
+        for (Map.Entry<String, Long> entry : partitionToSnapshotId.entrySet()) {
+            PaimonPartitionInfo partitionInfo = new PaimonPartitionInfo(entry.getKey(), entry.getValue());
+            result.add(partitionInfo);
+            if (entry.getValue() != null) {
+                latestSnapShotID = Math.max(latestSnapShotID, entry.getValue());
+            }
+        }
+        LOG.debug("Get changed partitionInfo:{}", result);
+        return result;
+    }
+
+    private Map<String, Long> fetchChangedPartitionWithVersion(PaimonTable paimonTable, long nextSnapshotId) {
```

Comment:
nextSnapshotId and latestSnapshotId are confused, they should be consistent

Reply:
done